pytensor/ipython.py

```diff
@@ -0,0 +1,202 @@
+import anywidget
+import ipywidgets as widgets
+import traitlets
+from IPython.display import display
+
+from pytensor.graph import FunctionGraph, Variable, rewrite_graph
+from pytensor.graph.features import FullHistory
+
+
+class CodeBlockWidget(anywidget.AnyWidget):
+    """Widget that displays text content as a monospaced code block."""
+
+    content = traitlets.Unicode("").tag(sync=True)
+
+    _esm = """
+    function render({ model, el }) {
```

Comment:
I think these should be put in a `widget.js` and a `widget.css` file instead of hanging out as a giant string.

Reply:
For now I would keep it together, until and if it actually grows larger. It's a single file this way instead of a folder. We can always refactor later